ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -252,3 +276,32 @@ private void removeAndAbort(GetOperation op, Exception abortCause) {
   }
 }
 
+/**
+ * Extension to {@link GetBlobOptions} containing additional parameters to the GetBlob operation that is internal to the
+ * router.
+ */
+class ExtendedGetBlobOptions extends GetBlobOptions {
```

Comment:
Discussed this offline with @vgkholla We concluded that this class is necessary. We also chose to make use of composition here. Made the change.